temporalio/failure.py

```diff
@@ -0,0 +1,61 @@
+from enum import IntEnum
+from typing import Any, List, Optional
+
+import temporalio.api.enums.v1
+import temporalio.api.failure.v1
+import temporalio.converter
+
+
+class FailureError(Exception):
+    @staticmethod
+    async def from_proto(
```

Comment:
Also don't love putting this in the public API

Reply:
Why not? What are we trying to hide? Are you afraid of this showing up in someone's autocomplete? Can you clarify your reasons on some of these comments?